athena-synapse/src/main/java/com/amazonaws/athena/connectors/synapse/SynapseMetadataHandler.java

```diff
@@ -143,59 +144,59 @@ public void getPartitions(BlockWriter blockWriter, GetTableLayoutRequest getTabl
         rowCountSt.add("name", getTableLayoutRequest.getTableName().getTableName());
         rowCountSt.add("schemaname", getTableLayoutRequest.getTableName().getSchemaName());
 
-        try (Connection connection = getJdbcConnectionFactory().getConnection(getCredentialProvider())) {
-            try (Statement st = connection.createStatement();
-                 Statement st2 = connection.createStatement();
-                 ResultSet resultSet = st.executeQuery(getPartitionsSt.render());
-                 ResultSet resultSet2 = st2.executeQuery(rowCountSt.render())) {
-                // check whether the table have partitions or not using ROW_COUNT_QUERY
-                if (resultSet2.next()) {
-                    rowCount = resultSet2.getInt("ROW_COUNT");
-                    LOGGER.info("rowCount: {}", rowCount);
-                }
-                // create a single split for view/non-partition table
-                if (rowCount == 0) {
-                    LOGGER.debug("Getting as single Partition: ");
-                    blockWriter.writeRows((Block block, int rowNum) ->
-                    {
-                        block.setValue(PARTITION_NUMBER, rowNum, ALL_PARTITIONS);
-                        //we wrote 1 row so we return 1
-                        return 1;
-                    });
-                }
-                else {
-                    LOGGER.debug("Getting data with diff Partitions: ");
+        try (Connection connection = getJdbcConnectionFactory().getConnection(getCredentialProvider());
+             Statement st = connection.createStatement();
+             Statement st2 = connection.createStatement();
+             ResultSet resultSet = st.executeQuery(getPartitionsSt.render());
+             ResultSet resultSet2 = st2.executeQuery(rowCountSt.render())) {
+            // check whether the table have partitions or not using ROW_COUNT_QUERY
+            if (resultSet2.next()) {
+                rowCount = resultSet2.getInt("ROW_COUNT");
+                LOGGER.info("rowCount: {}", rowCount);
+            }
+            // create a single split for view/non-partition table
+            if (rowCount == 0) {
```

Comment:
`rowCount` is only used by this block, so move the if == 0 if statement inside here and move the declaration of `rowCount` into here as well.